### ISSUE TITLE
LimitedSet.discard() fails to remove items from self._heap

### ISSUE THREAD

**Issue by djsmith42** [celery/celery#3087]
The `LimitedSet.discard()` method fails to remove items from `self._heap`. It looks like a simple typo:

```python
    def discard(self, value):
        """Remove membership by finding value."""
        try:
            itime = self._data[value]
        except KeyError:
            return
        try:
            self._heap.remove((value, itime)) # Bug here. These args are swapped.
        except ValueError as e:
            pass
        self._data.pop(value, None)
    pop_value = discard  # XXX compat
```

The call to `self._heap.remove()` should be like this:

```python
self._heap.remove((itime, value))
```

You can see in the `add()` method that items are added to the heap this way:

```python
heappush(self._heap, (inserted, key))
```

As a result, the LimitedSet has unbounded memory growth for callers who call `discard()` to remove items.

**Comment by djsmith42** [comment]
Pull request here: https://github.com/celery/celery/pull/3088

**Comment by ask** [comment]
Thanks! Merged into master and 3.1

**Comment by djsmith42** [comment]
Thank you @ask!
